substrate/rpc/src/state/mod.rs

```diff
@@ -93,18 +132,52 @@ impl<B, E, Block> StateApi<Block::Hash> for Arc<Client<B, E, Block>> where
 	}
 
 	fn storage_hash(&self, key: StorageKey) -> Result<Block::Hash> {
-		self.storage_hash_at(key, self.as_ref().info()?.chain.best_hash)
+		self.storage_hash_at(key, self.client.info()?.chain.best_hash)
 	}
 
 	fn storage_size(&self, key: StorageKey) -> Result<u64> {
-		self.storage_size_at(key, self.as_ref().info()?.chain.best_hash)
+		self.storage_size_at(key, self.client.info()?.chain.best_hash)
 	}
 
 	fn storage(&self, key: StorageKey) -> Result<StorageData> {
-		self.storage_at(key, self.as_ref().info()?.chain.best_hash)
+		self.storage_at(key, self.client.info()?.chain.best_hash)
 	}
 
 	fn call(&self, method: String, data: Vec<u8>) -> Result<Vec<u8>> {
-		self.call_at(method, data, self.as_ref().info()?.chain.best_hash)
+		self.call_at(method, data, self.client.info()?.chain.best_hash)
+	}
+
+	fn subscribe_storage(
+		&self,
+		_meta: Self::Metadata,
+		subscriber: pubsub::Subscriber<StorageChangeSet<Block::Hash>>,
+		keys: Trailing<Vec<StorageKey>>
+	) {
+		let keys = Into::<Option<Vec<_>>>::into(keys);
+		let stream = match self.client.storage_changes_notification_stream(keys.as_ref().map(|x| &**x)) {
+			Ok(stream) => stream,
+			Err(err) => {
+				let _ = subscriber.reject(error::Error::from(err).into());
+				return;
+			},
+		};
+
+		self.subscriptions.add(subscriber, |sink| {
+			let stream = stream
+				.map_err(|e| warn!("Error creating storage notification stream: {:?}", e))
+				.map(|(block, changes)| Ok(StorageChangeSet {
+					block,
+					changes: changes.iter().cloned().collect(),
+				}));
+			sink
+				.sink_map_err(|e| warn!("Error sending notifications: {:?}", e))
+				.send_all(stream)
+				// we ignore the resulting Stream (if the first stream is over we are unsubscribed)
```

Comment:
`…is over…`? Do you mean `…is closed`?

Reply:
`is over` as in `is finished/is done`. Which means that the stream will not emit any more items.